web-template/src/main/java/org/ehrbase/webtemplate/interpreter/Interpreter.java

```diff
@@ -0,0 +1,444 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ehrbase.aql.dto.condition.ConditionComparisonOperatorSymbol;
+import org.ehrbase.aql.dto.condition.SimpleValue;
+import org.ehrbase.aql.dto.containment.Containment;
+import org.ehrbase.aql.dto.containment.ContainmentDto;
+import org.ehrbase.aql.dto.containment.ContainmentExpresionDto;
+import org.ehrbase.aql.dto.path.AqlPath;
+import org.ehrbase.aql.dto.path.AqlPath.AqlNode;
+import org.ehrbase.aql.dto.path.predicate.PredicateComparisonOperatorDto;
+import org.ehrbase.aql.dto.path.predicate.PredicateLogicalAndOperation;
+import org.ehrbase.aql.dto.select.SelectFieldDto;
+import org.ehrbase.util.exception.SdkException;
+import org.ehrbase.webtemplate.model.WebTemplateInput;
+import org.ehrbase.webtemplate.model.WebTemplateNode;
+import org.ehrbase.webtemplate.templateprovider.TemplateProvider;
+
+/**
+ * @author Stefan Spiska
+ */
+public class Interpreter {
+
+    private final List<String> resolveTo;
+
+    private final TemplateProvider templateProvider;
+
+    /**
+     *
+     * @param templateProvider Used {@link TemplateProvider}
+     * @param resolveTo To wich {@link com.nedap.archie.rm.RMObject} should the part be resolved
+     */
+    public Interpreter(TemplateProvider templateProvider, List<String> resolveTo) {
+        this.templateProvider = templateProvider;
+        this.resolveTo = resolveTo;
+    }
+
+    /**
+     * Helper Method to convert the dots from the parser to one for the interpreter.
+     * @param selectFieldDto
+     * @param contains
+     * @return
+     */
+    private Set<InterpreterInput> toInterpreterInputSet(
+            SelectFieldDto selectFieldDto, ContainmentExpresionDto contains) {
+
+        return findContainment(selectFieldDto.getContainmentId(), contains).stream()
+                .map(r -> {
+                    InterpreterInput interpreterInput = new InterpreterInput();
+
+                    interpreterInput.setContainment(r.getRight());
+                    interpreterInput.setContainmentPath(
+                            Arrays.stream(r.getLeft()).collect(Collectors.toList()));
+                    interpreterInput.setPathFromContentment(selectFieldDto.getAqlPathDto());
+
+                    return interpreterInput;
+                })
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    public Set<InterpreterOutput> interpret(
+            SelectFieldDto selectFieldDto, ContainmentExpresionDto contains, String templateId) {
+
+        return interpret(toInterpreterInputSet(selectFieldDto, contains), templateId);
+    }
+
+    /**
+     * Called from {@link @see Interpreter#interpret(InterpreterInput, WebTemplateNode) }. Can be overwritten to take advantage of caching.
+     * @see Interpreter#interpret(InterpreterInput, WebTemplateNode)
```

Comment:
Method is not called from the indicated overloaded variant, it calls it. The only caller is `interpret(SelectFieldDto selectFieldDto, ContainmentExpresionDto contains, String templateId)`.